Drop assignments from sequences with no hits to equally-assigned sequences

eparejatobes opened the pull request:
See #48 for context.

If, after dropping inconsistent assignments, we have 
1. a sequence `S1` with assignment `T`
2. other sequences `S2,S3,...` which all have a `T` assignment
3. `S1` has no BLAST hits with `S2,S3,...`

Then the `T` assignment should be dropped from `S1`

Note that we can do this with input the MG7 results and the output assignments of the "drop inconsistent assignments" step.

eparejatobes committed:
add file for the drop no-hits step, concise definition

rtobes commented:
It could exist the case in which the T assignment is very unspecific (for example at phylum rank). In that case s1 assignment would be correct. I would apply that strategy only if T  is assigned to the **species** and perhaps also **genus** ranks.

eparejatobes commented:
https://github.com/ohnosequences/db.rna16s/pull/51#issue-172524590 is not exactly what I want, better check 7c73b03

eparejatobes commented:
> It could exist the case in which the T assignment is very unspecific (for example at phylum rank). In that case s1 assignment would be correct.

Yes, correct; but do we want to keep those assignments?

rtobes commented:
If they are the most specific assignment (obtained after automated curation) for that sequences perhaps yes

eparejatobes committed:
add implementation sketch for the drop no hits step

eparejatobes commented:
> If they are the most specific assignment (obtained after automated curation) for that sequences perhaps yes

If they were assigned to something like Firmicutes, Bacteria, etc then it probably was through (weak) similarity with well identified sequences. Then, I prefer to keep only those well identified sequences, and if I happen to get that sequence as input when running mg7 with this DB, the result will be equivalent, if not better: most likely we will have more well-identified sequences than those which where used when classifying that sequence.

rtobes commented:
My concern is about really strange and new sequences not similar to nothing in the database. 
Even without a filiation, it would be interesting to know that you have a sequence similar to one in the database even when you do not know what it is. 
Perhaps we can add new "taxaIDs" to include the "not classified unique sequences" with no similar ones in the database of well classified sequences

Commit:
I have highlighted the line 125 because it summarizes all the process 

Here we discard (from the query sequence data) those taxa assigned whose lineage does **not** contain the *parent* of the lca assignment (obtained in the MG7 test).

eparejatobes commented:
After discussing this with @rtobes we have decided to leave this on hold. It's not really well-defined, and there's a high risk of dropping interesting assignments/sequences.
